src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -477,9 +505,47 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
         {
             // Called outside of async state machine to propagate certain exception even without awaiting the returned task.
             CheckRequestBeforeSend(request);
-
             (CancellationTokenSource cts, bool disposeCts, CancellationTokenSource pendingRequestsCts) = PrepareCancellationTokenSource(cancellationToken);
-            return SendAsyncCore(request, completionOption, async: true, cts, disposeCts, pendingRequestsCts, cancellationToken).AsTask();
+
+            return Core(request, completionOption, cts, disposeCts, pendingRequestsCts, cancellationToken);
+
+            async Task<HttpResponseMessage> Core(
```

Comment:
Why an extra method? Why not just have the code as a body of `SendAsync`?

Reply:
So that argument exceptions from CheckRequestBeforeSend propagate synchronously.

Reply:
I see. So why the change from private method to a local one? Are there any benefits, like perf/memory/trimming? I'm just curious.

Reply:
> I see. So why the change from private method to a local one? Are there any benefits, like perf/memory/trimming? I'm just curious.

Zero functional/performance benefit; the compiler will generate a private method to implement it.  The benefit is purely about code maintenance and what code should be able to see this function: just how you'd make a method private instead of internal so that only the containing method could see it, you can make a method local so that only the containing method can see it.